src/palace/manager/api/admin/controller/work_editor.py

```diff
@@ -367,45 +368,58 @@ def edit(self, identifier_type, identifier):
 
         return Response("", 200)
 
-    def suppress(self, identifier_type, identifier):
-        """Suppress the license pool associated with a book."""
-        self.require_librarian(flask.request.library)
+    def suppress(
+        self, identifier_type: str, identifier: str
+    ) -> Response | ProblemDetail:
+        """Suppress a book at the level of a library."""
+
+        library = flask.request.library  # type: ignore[attr-defined]
+
+        if library is None:
+            return self.no_library_response()
```

Comment:
This is fine, though it returns a 400. In other places (e.g., [inventory reports](https://github.com/ThePalaceProject/circulation/blob/d213661bd86185ad32127410837e6526f12474ad/src/palace/manager/api/admin/controller/report.py#L35-L37) and [the `has_library` decorator via `BaseCirculationController.library_for_request`](https://github.com/ThePalaceProject/circulation/blob/85615864fea7bf0906fd1321dbf2f06cf0602065/src/palace/manager/api/controller/base.py#L130-L131)), we return the `LIBRARY_NOT_FOUND` problem detail or just raise a ProblemDetailException with it. These return 404.

```python
...
from palace.manager.api.problem_details import LIBRARY_NOT_FOUND
...
    library: Library | None = getattr(request, "library")
    if library is None:
        raise ProblemDetailException(LIBRARY_NOT_FOUND)
```

Reply:
I actually think that this should return a 404, but fine to leave this as-is here and I can update in the in-progress follow-up PR.